sxg_rs/src/acme/jws.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module implements
+//! [JSON Web Signature (JWS)](https://datatracker.ietf.org/doc/html/rfc7515).
+//! [ACME spec](https://datatracker.ietf.org/doc/html/rfc8555#:~:text=payload%20in%20a%20JSON%20Web%20Signature)
+//! requires the request body to be encapsulated in JWS format for authenication.
+
+use crate::crypto::EcPublicKey;
+use crate::signature::Signer;
+use anyhow::{Error, Result};
+use serde::Serialize;
+
+pub async fn create_acme_request_body<S: Signer, P: Serialize>(
```

Comment:
Where is this called?

Reply:
This PR only focuses on JWS. A following up PR of ACME will use this function.